### ISSUE TITLE
docker image bitwardenrs/server-mysql never uses mysql

### ISSUE THREAD

**Issue by maoxuner** [dani-garcia/bitwarden_rs#512]
I have corrected set the envars accordding to wiki [Using the MySQL Backend](https://github.com/dani-garcia/bitwarden_rs/wiki/Using-the-MySQL-Backend), and bitwarden_rs never connects mysql but always create a sqlite file by DATABASE_URL like `mysql://user:pass@mysql/user_bitwarden`（My database container 's name is `mysql`）.

```
root@bitwarden:/# file mysql\:/user\:pass\@mysql/user_bitwarden 
mysql:/user:pass@mysql/user_bitwarden: SQLite 3.x database, last written using SQLite version 3026000
```

Part of the docker-compose file
```
version: "3"
serevices:
  bitwarden:
    container_name: bitwarden
    hostname: bitwarden
    image: bitwardenrs/server-mysql:${BITWARDEN_TAG:-latest}
    deploy:
      restart_policy:
        condition: any
        delay: 5s
    restart: always
    environment:
      WEBSOCKET_ENABLED: "true"
      SIGNUPS_ALLOWED: "true"
      ADMIN_TOKEN: admin
      DATABASE_URL: "mysql://user:pass@mysql/user_bitwarden"
      ENABLE_DB_WAL: "false"
    networks:
      - underlay
      - database
networks:
  underlay:
  database:
```

I tried build the image for myself following [Building your own image](https://github.com/dani-garcia/bitwarden_rs/wiki/Building-your-own-image).
```
docker build -t bitwarden_rs --build-arg DB=mysql .
```
Then it works well. So I suppose there may be a wrong build argument with the image `bitwardenrs/server-mysql`.

**Comment by mprasil** [comment]
It seems like we're victims of incorrect/confusing name on the build setup page:
![image](https://user-images.githubusercontent.com/8255364/60003396-8131d980-9662-11e9-8754-01ce67e35366.png)

[Apparently](https://github.com/docker/hub-feedback/issues/508#issuecomment-493630920) this setting on the build configuration screen does not set `--build-arg`, but it sets environment variable? But `docker build` has no `--env` option, so I have no idea what that thing actually does..

We might have to either setup custom hooks to push the ARG that way or create separate version of Dockerfile for MySQL. (which sounds like more transparent approach)

**Comment by jtcressy** [comment]
Is this relevant, or has it already been tried? https://github.com/docker/hub-feedback/issues/508#issuecomment-222520720

TL;DR:
In the repository root:
`touch hooks/build`
Contents:
```
#!/bin/sh
docker build --build-arg DB=mysql -t $IMAGE_NAME .
```

**Comment by paulolieuthier** [comment]
Latest tag of bitwardenrs/server-mysql, 1.10.0, still does not use MySQL, right?

**Comment by ohkeenan** [comment]
> Latest tag of bitwardenrs/server-mysql, 1.10.0, still does not use MySQL, right?

Not sure what you mean exactly? 1.10.0 [dockerfile](https://hub.docker.com/r/bitwardenrs/server-mysql/dockerfile) installs `libmariadb-dev` and `libmariadbclient-dev` which are interchangeable with MySQL.

Here's an example docker-compose.yml:
```version: "3"

services:
  db:
    image: mysql
    command: --default-authentication-plugin=mysql_native_password #this might not be necessary
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: CHANGETHIS1234
      MYSQL_DATABASE: bitwarden
      MYSQL_USER: bitwarden
      MYSQL_PASSWORD: CHANGETHIS123
    volumes:
      - db:/var/lib/mysql

  bitwarden:
    image: bitwardenrs/server-mysql
    restart: always
    volumes:
      - ./bw-data:/data
    ports:
    - 80:80
    - 3012:3012
    depends_on:
      - db
    environment:
      WEBSOCKET_ENABLED: "true" # Required to use websockets
      SIGNUPS_ALLOWED: "true" # set to false to disable signups
      DATABASE_URL: "mysql://bitwarden:CHANGETHIS123@db/bitwarden"
      ENABLE_DB_WAL: "false"

volumes:
  db:
```

**Comment by paulolieuthier** [comment]
In your example you pull the `latest` tag. My point is that the tag `1.10.0` is still not using MySQL.

**Comment by ohkeenan** [comment]
> In your example you pull the `latest` tag. My point is that the tag `1.10.0`, which is the latest "stable" tag (I don't want to be using `latest`) is still not using MySQL.

Ahh yeah you're absolutely right adding 1.10.0 to image version appears to not use the SQL database at least in my testing.

docker logs warden_bitwarden_1
```
[2019-09-28 22:33:07][rocket::rocket][INFO] POST /api/accounts/register application/json; charset=utf-8:
[2019-09-28 22:33:07][_][INFO] Matched: POST /api/accounts/register (register)
[2019-09-28 22:33:07][_][INFO] Outcome: Success
[2019-09-28 22:33:07][_][INFO] Response succeeded.
[2019-09-28 22:33:10][rocket::rocket][INFO] POST /api/accounts/prelogin application/json; charset=utf-8:
[2019-09-28 22:33:10][_][INFO] Matched: POST /api/accounts/prelogin (prelogin)
[2019-09-28 22:33:10][_][INFO] Outcome: Success
[2019-09-28 22:33:10][_][INFO] Response succeeded.
[2019-09-28 22:33:10][rocket::rocket][INFO] POST /identity/connect/token application/x-www-form-urlencoded; charset=utf-8:
[2019-09-28 22:33:10][_][INFO] Matched: POST /identity/connect/token (login)
[2019-09-28 22:33:10][bitwarden_rs::api::identity][INFO] User ****** logged in successfully. IP: 172.19.0.1
[2019-09-28 22:33:10][_][INFO] Outcome: Success
[2019-09-28 22:33:10][_][INFO] Response succeeded.
```

docker exec -it warden_db_1 /bin/bash
mysql -uroot -p
```
mysql> select * from bitwarden.users;
Empty set (0.00 sec)```


It also creates a directory in / I think is a mistake? This doesn't happen with :latest
```
~/Development/warden on 🐳 v19.03.2 
➜ docker exec -it warden_bitwarden_1 /bin/bash
root@ba6a69aeee52:/# ls
Rocket.toml  bitwarden_rs  data  etc   lib    media  mysql:  proc  run	 srv  tmp  var
bin	     boot	   dev	 home  lib64  mnt    opt     root  sbin  sys  usr  web-vault
root@ba6a69aeee52:/# ls -lah mysql\:/bitwarden\:CHANGETHIS123\@db/bitwarden 
-rw-r--r-- 1 root root 136K Sep 28 22:33 mysql:/bitwarden:CHANGETHIS123@db/bitwarden
```

**Comment by paulolieuthier** [comment]
Yeah, that's right. Pushing another tag to Dockerhub is enough to fix it (and please reopen this while that's not done).

While I'm at it: it would be nice if stable tags were recommended instead of pulling latest. That way upgrading would be intentional, with fewer surprises, and you guys would communicate features and fixes explicitly to your users (btw, thanks for your work!).
